### PR TITLE
Capture JSON RPC errors that presently get swallowed

### LINKED ISSUE #28531: json_rpc_2 is swallowing vmservice's exceptions
See https://github.com/flutter/flutter/pull/28530 and https://github.com/flutter/flutter/issues/28374

The exceptions are currently swallowed in https://github.com/dart-lang/json_rpc_2/blob/master/lib/src/server.dart#L178

### PR DESCRIPTION
**Pull request by dnfield**
## Description

json_rpc_2 swallows exceptions. This can hide bugs in our callbacks, e.g. type conversion issues, which can result in an RPC call not doing what is expected.

This change hooks into a new parameter available that informs the caller of unhandled exceptions.  For now, we just print them and assert(false).  For Fuchsia debug protocol, we only print them.

## Related Issues

Fixes #28531 

## Tests

The tests for this functionality exist upstream in the PR I wrote for json_rpc_2.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter_driver/lib/src/driver/driver.dart [hunk 1]
```diff
@@ -933,6 +933,10 @@ void restoreVmServiceConnectFunction() {
   vmServiceConnectFunction = _waitAndConnect;
 }
 
+void _unhandledJsonRpcError(dynamic error, dynamic stack) {
+
```
Comment (liyuqian) [inline packages/flutter_driver/lib/src/driver/driver.dart]: Is this intentionally blank?

Comment (dnfield) [reply packages/flutter_driver/lib/src/driver/driver.dart]: No, forgot to save.  Will fix tonight

**Comment by dnfield** [comment]
@jonahwilliams any idea on the  builderOptions failures?  Do we need to pin build_runner?

**Comment by jonahwilliams** [comment]
We can pin it for now and I can handle rolling the changes

File: packages/flutter_tools/lib/src/commands/update_packages.dart [hunk 2]
```diff
@@ -23,6 +23,7 @@ import '../runner/flutter_command.dart';
 const Map<String, String> _kManuallyPinnedDependencies = <String, String>{
   // Add pinned packages here.
   'flutter_gallery_assets': '0.1.8', // See //examples/flutter_gallery/pubspec.yaml
+  'build_modules': '1.0.9', // TODO(jonahwilliams): migrate https://cirrus-ci.com/task/5747941259083776
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/commands/update_packages.dart]: Thanks!

**Review by jonahwilliams — APPROVED** [review]
LGTM

**Comment by dnfield** [comment]
This spiked our tech debt benchmark for package deps, but didn't add any new deps...

**Comment by dnfield** [comment]
It's because of the pinned dependency.  That will go back down once we unpin the dep.

**Pull request merged**